app/src/main/java/com/github/se/travelpouch/ui/dashboard/NextActivitiesBanner.kt

```diff
@@ -38,56 +47,86 @@ import com.google.firebase.Timestamp
 fun NextActivitiesBanner(
     activities: List<Activity>, // Your activity data model
     onDismiss: () -> Unit,
-    modifier: Modifier = Modifier
+    modifier: Modifier = Modifier,
+    localConfig: Configuration = LocalConfiguration.current
 ) {
+  var isPortrait = localConfig.orientation == ORIENTATION_PORTRAIT
+  var screenSize = if (isPortrait) localConfig.screenHeightDp else localConfig.screenWidthDp
+  val dismissButtonSize = (screenSize * 0.05).dp // 5% of the screen size
   val showOutline = remember { mutableStateOf(false) }
   // Filter activities due within the next 24 hours
   val reminderRange = 86400L // 2400 hours in milliseconds
   val nowTime = Timestamp.now().seconds
+
   val upcomingActivities =
-      activities.filter { activity ->
-        val timeDiff: Long = activity.date.seconds - nowTime
-        timeDiff in 0..reminderRange // 24 hours in milliseconds
-      }
+      activities
+          .filter { activity ->
+            val timeDiff: Long = activity.date.seconds - nowTime
+            timeDiff in 0..reminderRange // 24 hours in milliseconds
+          }
+          .sortedBy { it.date } // Sort by due time
 
   if (upcomingActivities.isNotEmpty()) {
     Box(
         modifier =
             modifier
-                .background(Color.Gray.copy(alpha = 0.8f), RoundedCornerShape(8.dp))
-                .clickable {
-                  // Set the outline on first tap
-                  showOutline.value = true
-                }
-                .padding(16.dp)
+                .heightIn(
+                    max =
+                        (0.3f * LocalConfiguration.current.screenHeightDp.dp.value)
+                            .dp) // 30% of screen height
+                .background(Color.Gray.copy(alpha = 0.85f), RoundedCornerShape(8.dp))
+                .clickable { showOutline.value = true } // Set the outline on first tap
+                .padding(horizontal = 16.dp, vertical = 8.dp)
```

Comment:
We don't do this anywhere else in our code, this would set a very bad precedent. While these are "magic values", they are just arbitrary alignments values and hold no deeper meaning.